### PR TITLE
Feat: ORV2-1500 another update on company search

### PR DESCRIPTION
**Pull request by erikataot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1101-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1101-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1101-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1101-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

**Review by praju-aot — APPROVED** [review]
Approved backend change

File: frontend/src/features/idir/search/table/CompanySearchResultColumnDef.tsx [hunk 1]
```diff
@@ -26,8 +26,8 @@ export const CompanySearchResultColumnDef: MRT_ColumnDef<CompanyProfile>[] = [
     sortingFn: "alphanumeric",
   },
   {
-    accessorKey: "migratedClientHash",
-    header: "Client Number",
+    accessorKey: "clientNumber",
```
Comment (krishnan-aot) [inline frontend/src/features/idir/search/table/CompanySearchResultColumnDef.tsx:29]: Aren't we negotiating the clientNumber as a hash?

Comment (erikataot) [reply frontend/src/features/idir/search/table/CompanySearchResultColumnDef.tsx:29]: in the `read-company` DTO from the service we're supplying both:
```
  @AutoMap()
  @ApiProperty({
    description: 'The ORBC client number.',
    example: '1234',
  })
  clientNumber: string;

  @AutoMap()
  @ApiProperty({
    description: 'The migrated client hash.',
    example: '8db3c29a00c728c0a3c31b91662a4ba4280e50ac6bfc388a7765d1bcb46845da',
    required: false,
  })
  migratedClientHash?: string;
  ```

Comment (krishnan-aot) [reply frontend/src/features/idir/search/table/CompanySearchResultColumnDef.tsx:29]: ok, thanks!

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**